api/v1/egressgateway_types.go

```diff
@@ -65,6 +65,12 @@ type EgressGatewaySpec struct {
 	// +required
 	IPPools []EgressGatewayIPPool `json:"ipPools"`
 
+	// ExternalNetworks defines the external network names this Egress Gateway is
+	// associated with.
+	// ExternalNetworks must match existing external networks.
+	// +optional
+	ExternalNetworks []string `json:"externalNetworks,omitempty"`
```

Comment:
omitempty was a good catch - we should put that on all of the optional fields. Sorry for missing that in the original review.

Reply:
updated omitempty